src/Bicep.Local.Extension/Protocol/IResourceHandler.cs

```diff
@@ -4,53 +4,57 @@
 using System.Collections.Immutable;
 using System.Text.Json.Nodes;
 using System.Text.Json.Serialization.Metadata;
+using Bicep.Local.Extension.Rpc;
 
 namespace Bicep.Local.Extension.Protocol;
 
-public record ExtensibilityOperationRequest(
-    ExtensibleImportData Import,
-    ExtensibleResourceData Resource);
-
-public record ExtensibilityOperationResponse(
-    ExtensibleResourceData? Resource,
-    ExtensibleResourceMetadata? ResourceMetadata,
-    ImmutableArray<ExtensibilityError>? Errors);
-
-public record ExtensibleImportData(
-    string Provider,
-    string Version,
-    JsonObject? Config);
+public record ResourceRequestBody(
+    JsonObject? Config,
+    string Type,
+    JsonObject Properties,
+    string? ApiVersion);
 
-public record ExtensibleResourceData(
+public record ResourceReferenceRequestBody(
+    JsonObject Identifiers,
+    JsonObject? Config,
     string Type,
-    JsonObject? Properties);
+    string? ApiVersion);
 
-public record ExtensibleResourceMetadata(
-    ImmutableArray<string>? ReadOnlyProperties,
-    ImmutableArray<string>? ImmutableProperties,
-    ImmutableArray<string>? DynamicProperties);
+public record ResourceResponseBody(
+    ErrorPayload? Error,
+    JsonObject Identifiers,
+    string Type,
+    string Status,
+    JsonObject Properties);
 
-public record ExtensibilityError(
+public record ErrorPayload(
     string Code,
+    string Target,
     string Message,
-    string Target);
+    ErrorDetail[]? Details,
+    JsonObject? InnerError);
```

Comment:
Is there an expected contract for the `InnerError` field?

Reply:
Yes, that's defined by the OData 4.0 error response format: https://docs.oasis-open.org/odata/odata-json-format/v4.0/os/odata-json-format-v4.0-os.html#_Toc372793091, which is used by the Graph API and some Azure data plane APIs. The deployment engine converts it to `error.details[*].additionalInfo` to ensure compatibility with ARM RPC.

Reply:
Thanks - so sounds like using `JsonObject` rather than defining a custom `record` for it is the way to go?

Reply:
Yup